graders/python/python_autograder/pltest.py

```diff
@@ -0,0 +1,127 @@
+from unittest import TestLoader, TestSuite
+import json
+import traceback
+import os
+from collections import defaultdict
+from pl_result import PrairieTestResult
+
+
+"""
+The main python entrypoint for the autograder framework.
+Loads and executes test cases.
+"""
+
+
+def add_files(results):
+    BASE_DIR = "/grade/run/"
+
+    for test in results:
+        test["files"] = test.get("files", [])
+        image_fname = BASE_DIR + "image_" + test["name"] + ".png"
+        if os.path.exists(image_fname):
+            with open(image_fname, 'r') as content_file:
+                imgsrc = content_file.read()
+            test["files"].append({"name":"image_" + test["filename"] + ".png",
+                                  "imgsrc":imgsrc})
+            os.remove(image_fname)
+        feedback_fname = BASE_DIR + "feedback_" + test["filename"] + ".txt"
+        if os.path.exists(feedback_fname):
+            with open(feedback_fname, 'r', encoding='utf-8') as content_file:
+                text_output = content_file.read()
+            test["files"].append({"name":"feedback_" + test["filename"] + ".txt",
+                                  "text_output":text_output})
+            os.remove(feedback_fname)
+
```

Comment:
@bowmnath do you have any thoughts on simplifying the `files` in test cases?  I wonder if we could just have `output` and `images` keys but I don't want to break anything -- also fine just leaving it as is since I feel like this pr has already dragged on long enough :slightly_smiling_face:

Reply:
To be honest, similar code was there when I started writing the autograder, and I was hesitant to do anything to it because I didn't completely understand how the results were used.  If you have a good idea for how to handle this, go for it.  Just looking through `pltest.py`, I can't immediately see how the files are used.  Must be in the external results element?  In any case, you've put in a ton of work here, and if this is the primary thing holding up the PR, discretion might be the better part of valor on this one. :slightly_smiling_face:

Reply:
Gotcha, I think I'd rather refactor this so that it makes sense to anyone else using the results element.